### DISCUSSION TITLE
Filepond upload loses original filename

### DISCUSSION THREAD

**Discussion by robertorinaldi-dev** [livewire/livewire#4464] [7. Bugs (archived)]
I have this problem as well https://github.com/livewire/livewire/issues/2918.
For reasons I don't know, when I upload with filepond (using filament), the original name become 'blob'.
Adding file.name to row 71 of UploadManager.js seems to solve it
```php
Array.from(this.uploadBag.first(name).files).forEach(file => formData.append('files[]', file, file.name))
```
Was there a reason not to include it?

I would make a PR but don't know why running phpunit I get various errors that have nothing to do with it and it crashes when running dusk.

Thanks

**Comment by robertorinaldi-dev** [comment]
Cc @danharrin

**Comment by robertorinaldi-dev** [comment]
Ok, now I know why I get 'blob' as filename: https://developer.mozilla.org/en-US/docs/Web/API/FormData/append

**Reply by joshhanley** [reply]
@robertorinaldi-dev did you manage to sort this out? Does using `getClientOriginalName()` work?

**Reply by danharrin** [reply]
A PR is pending - #4486

**Reply by joshhanley** [reply]
@danharrin thanks! Yeah I commented on the PR 🙂
